pkg/bot/slack_cloud.go

```diff
@@ -108,21 +112,27 @@ func NewCloudSlack(log logrus.FieldLogger,
 		clusterName:      clusterName,
 		realNamesForID:   map[string]string{},
 		msgStatusTracker: NewSlackMessageStatusTracker(log, client),
+		status:           StatusUnknown,
+		maxRetries:       maxRetries,
+		failuresNo:       0,
+		failureReason:    "",
```

Comment:
All reasons work good. However, I got the `Unknown` status until I run at least one command from Slack. The reason is that you have `setFailureReason` in `handleStreamMessage` and the `b.start` is long-running function, exits only on error so the `setFailureReason` from `retry.Do` doesn't work as expected. In general, you should fix it so when started and there are no errors, the status is `Healthy`.

Reply:
fixed